server/nodeset/upload-deposit-data.go

```diff
@@ -132,25 +128,21 @@ func (c *nodesetUploadDepositDataContext) PrepareData(data *swapi.NodesetUploadD
 	}
 	data.Balance = balance
 
-	totalCost := new(big.Int)
+	totalCost := big.NewInt(0)
+	costPerKey := eth.EthToWei(validatorDepositCost)
 
-	// Add deposit cost for each unregistered key
-	for range unregisteredKeys {
-		totalCost.Add(totalCost, eth.EthToWei(validatorDepositCost))
-	}
-	// Total cost needs to account for pending validators
-	for range pendingPubkeysOnNodeset {
-		totalCost = totalCost.Add(totalCost, eth.EthToWei(validatorDepositCost))
-	}
+	unregisteredKeysCount := len(unregisteredKeys)
+	pendingPubkeysOnNodesetCount := len(pendingPubkeysOnNodeset)
+
+	totalCost.Add(totalCost, new(big.Int).Mul(costPerKey, big.NewInt(int64(unregisteredKeysCount+pendingPubkeysOnNodesetCount))))
```

Comment:
Recommend putting the 2nd portion (the `Mul`) on its own line above for clarity, and use `big.NewInt(0)` instead of `new(big.Int)` for consistency.

Reply:
resolved